S3/Config.py

```diff
@@ -318,7 +318,16 @@ def role_config(self):
                     '%s=%s' % (k, s3_quote(v, unicode_output=True))
                     for k, v in params.items()
                 ])
-                conn = httplib.HTTPSConnection(host='sts.amazonaws.com',
+                sts_endpoint = "sts.amazonaws.com"
+                if os.environ.get("AWS_STS_REGIONAL_ENDPOINTS") == "regional": 
+                    # Check if the AWS_REGION variable is available to use as a region. 
+                    if os.environ.get("AWS_REGION"): 
+                        region = os.environ.get("AWS_REGION")
```

Comment:
Instead of getting the value 2 times, you can do:
```
region = os.env...
if not region:
    region = self.bucket_location
```

Reply:
Done thanks for the feedback !